app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningFragment.kt

```diff
@@ -94,18 +98,39 @@ class LearningFragment : Fragment() {
             }
             targetImageViews = updateTargetImageViews()
             learningPresenter =
-                LearningPresenter(dbMgt, args.learningMode, dataset, authPresenter, imageDisplayer)
+                LearningPresenter(
+                    dbMgt, args.learningMode, dataset, authPresenter, imageDisplayer,
+                    lifecycleScope
+                )
             learningPresenter.updateForNextSorting(
-                parentActivity,
                 targetImageViews,
                 binding.learningToSort
             )
             // sets the listeners for the image views of the sorting
             setEventListeners()
         }
+
+        visualFeedback = with(binding) {
+            LearningVisualFeedback(
+                lifecycleScope,
+                baseColor = getColor(requireContext(), R.color.white),
+                neutralColor = getColor(requireContext(), R.color.blue),
+                positiveColor = getColor(requireContext(), R.color.light_green),
+                negativeColor = getColor(requireContext(), R.color.red),
+                // TODO remove !! once enclosing CardViews put in all layouts
+                sourceCardView = learningToSortCv!!,
```

Comment:
I don't perfecly understand this TODO. I mean what did you want to do to remove this todo?

Reply:
I haven't added the cardview behind the image in all versions of the learning layout, so if you I need to put `!!` to use the cardviews. This is bad because the learning will crash (with a NullPointerException)  if you use it on a tablet for example. But I didn't have the time to adjust all the layouts...

Reply:
oooh got it thanks